community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/mutation/DeleteEntityAction.scala

```diff
@@ -24,31 +24,35 @@ import org.neo4j.cypher.internal.compiler.v2_3.commands.expressions.{Expression,
 import org.neo4j.cypher.internal.compiler.v2_3.executionplan.{Effects, _}
 import org.neo4j.cypher.internal.compiler.v2_3.pipes.QueryState
 import org.neo4j.cypher.internal.compiler.v2_3.symbols._
-import org.neo4j.graphdb.{Node, Path, PropertyContainer, Relationship}
+import org.neo4j.graphdb
 
 import scala.collection.JavaConverters._
 
-case class DeleteEntityAction(elementToDelete: Expression)
+case class DeleteEntityAction(elementToDelete: Expression, forced: Boolean)
   extends UpdateAction {
 
   def exec(context: ExecutionContext, state: QueryState) = {
     elementToDelete(context)(state) match {
-      case n: Node => delete(n, state)
-      case r: Relationship => delete(r, state)
+      case n: graphdb.Node => delete(n, state, forced)
+      case r: graphdb.Relationship => delete(r, state, forced)
       case null =>
-      case p:Path => p.iterator().asScala.foreach( pc => delete(pc, state))
+      case p: graphdb.Path => p.iterator().asScala.foreach(pc => delete(pc, state, forced))
       case x => throw new CypherTypeException("Expression `" + elementToDelete.toString() + "` yielded `" + x.toString + "`. Don't know how to delete that.")
     }
     Iterator(context)
   }
 
-  private def delete(x: PropertyContainer, state: QueryState) {
-
+  private def delete(x: graphdb.PropertyContainer, state: QueryState, forced: Boolean) {
```

Comment:
could forced default to false? so you don't have to change all the places where it is called?